src/wrappers.jl

```diff
@@ -0,0 +1,442 @@
+end
+function Feature(geometry=nothing; properties=nothing, crs=nothing, extent=nothing)
+    if isnothing(geometry) || isgeometry(geometry)
+        # Wrap a NamedTuple feature
+        Feature((; geometry, properties...), crs, extent)
+    else
+        throw(ArgumentError("object must be a feature, geometry or `nothing`. Got $(typeof(geometry))"))
+    end
+end
+
+Base.parent(f::Feature) = f.parent
+
+isfeature(::Type{<:Feature}) = true
+trait(::Feature) = FeatureTrait()
+geometry(f::Feature) = geometry(parent(f))
+properties(f::Feature) = properties(parent(f))
+extent(f::Feature) =
+    isfeature(parent(f)) && isnothing(f.extent) ? extent(parent(f)) : f.extent
+crs(f::Feature) =
+    isfeature(parent(f)) && isnothing(f.crs) ? crs(parent(f)) : f.crs
+
+"""
+    FeatureCollection(features; [crs, extent])
+
+A FeatureCollection wrapper.
+
+## Arguments
+
+- `features`: an `AbstractArray` of GeoInterface compatible features.
+    Iterables are accepted but will be collected to an `Array`.
+
+## Keywords
+
+- `crs`: Any GeoFormatTypes.jl crs type, or `nothing`
+- `extent`: An Extents.jl `Extent` or `nothing`
+
+## Examples
+
+```julia
+fc = FeatureCollection([feature1, feature2, feature3];
+    crs=EPSG(4326), extent=Extent(X=(11.0, 34.0), Y=(45.7, 78.0))
+)
+```
+
+Or wrap another `FeatureColection`, e.g. if it has no crs attached:
+
+```julia
+fc = FeatureCollection(featurecollection, crs=EPSG(4326))
+```
+"""
+struct FeatureCollection{P,C,E}
+    parent::P
+    crs::C
+    extent::E
+end
+function FeatureCollection(fc::FeatureCollection; crs=crs(fc), extent=extent(fc))
+    FeatureCollection(parent(fc), crs, extent)
+end
+function FeatureCollection(parent; crs=nothing, extent=nothing)
+    if isfeaturecollection(parent)
+        FeatureCollection(parent, crs, extent)
+    else
+        features = (parent isa AbstractArray) ? parent : collect(parent) 
+        all(f -> isfeature(f), features) || _child_feature_error()
+        FeatureCollection(parent, crs, extent)
+    end
+end
+
+Base.parent(fc::FeatureCollection) = fc.parent
+
+_child_feature_error() = throw(ArgumentError("child objects must be features"))
+
+isfeaturecollection(fc::Type{<:FeatureCollection}) = true
+trait(fc::FeatureCollection) = FeatureCollectionTrait()
+
+function nfeature(::FeatureCollectionTrait, fc::FeatureCollection)
+    isfeaturecollection(parent(fc)) ? nfeature(t, parent(fc)) : length(parent(fc))
+end
+getfeature(::FeatureCollectionTrait, fc::FeatureCollection) =
+    isfeaturecollection(parent(fc)) ? getfeature(t, parent(fc)) : parent(fc)
+getfeature(t::FeatureCollectionTrait, fc::FeatureCollection, i::Integer) =
+    isfeaturecollection(parent(fc)) ? getfeature(t, parent(fc), i) : parent(fc)[i]
+extent(fc::FeatureCollection) =
+    (isnothing(fc.extent) && isfeaturecollection(parent(fc))) ? extent(parent(fc)) : fc.extent
+crs(fc::FeatureCollection) =
+    isfeaturecollection(parent(fc)) && isnothing(fc.crs) ? crs(parent(fc)) : fc.crs
```

Comment:
These all repeat the `isfeaturecollection(parent(fc))` check, which is also done in the constructor. Might be good to generalize.

Reply:
Ok I'll may a method